lib/domain/orders_repository.dart

```diff
@@ -0,0 +1,71 @@
+import 'dart:convert';
+import 'dart:io';
+import 'package:mvp/classes/prefrenses.dart';
+import 'package:mvp/constants/apiCalls.dart';
+import 'package:mvp/domain/exceptions.dart';
+import 'package:mvp/models/ordersModel.dart';
+import 'package:http/http.dart' as http;
+
+abstract class OrdersRepository {
+  Future<List<OrderModel>> fetchOrders();
+}
+
+class OrdersRepositoryImpl implements OrdersRepository {
+  Future _fetch() async {
+    try {
+      final p = await Preferences.getInstance();
+      String id = await p.getData("id");
+      String token = await p.getData("token");
+      String url = APIService.ordersAPI + "$id";
```

Comment:
the only change here is the url, everything else is same as in other fetch functions. why not make an abstract class and then derive it ?

Reply:
will do it